### PR TITLE
Fix nested notations vs ltac2

### LINKED ISSUE #15864: Notation cannot use another notation containing ltac2-in-term
```coq
Require Import Ltac2.Ltac2.

(* This line is copied from https://coq.inria.fr/refman/proof-engine/ltac2.html *)
Notation "[ x ]" := ltac2:(let x := Ltac2.Constr.pretype x in exact $x) (only parsing).

Check [ 1 ].

Notation "<< y >>" := [ y ] (only parsing).

Check << 1 >>.
```

The second `Check` results in `Error: The variable y was not found in the current environment.`

Expected behavior: The second `Check` should succeed like the first one.

Coq version: 8.15-rc1

/cc @ppedrot

### PR DESCRIPTION
**Pull request by SkySkimmer**
Fix #15864

### CODE REVIEW

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by jfehrle** [comment]
"Imbricated"??  Do you mean just "overlapping"?  I've never seen the word before.  I expect many native English speakers don't know it.

The usage example from the dictionary is amusingly opaque:

overlap or cause to overlap.
"a distinguishing feature of the echinoids is that the ossicles imbricate". 😊

**Comment by proux01** [comment]
Or maybe "nested" (we usually translate the French "boucles imbriquées" as "nested loops").

**Comment by SkySkimmer** [comment]
Nested is good

**Comment by SkySkimmer** [comment]
@coqbot run full ci

File: doc/changelog/06-Ltac2-language/17232-tac2-preterm-notations.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+- **Fixed:**
+  imbricated notations involving :ref:`term-antiquotations`
```
Comment (ppedrot) [inline doc/changelog/06-Ltac2-language/17232-tac2-preterm-notations.rst:2]: This should read *nested*.

Comment (SkySkimmer) [reply doc/changelog/06-Ltac2-language/17232-tac2-preterm-notations.rst]: changed

**Comment by SkySkimmer** [comment]
Added fix for https://github.com/coq/coq/issues/17233

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**